tests/node/test_wakunode_peer_manager.nim

```diff
@@ -577,77 +578,52 @@ suite "Peer Manager":
-          check:
-            clientPeerStore.get(serverPeerId).connectedness == Connectedness.CanConnect
-            serverPeerStore.get(clientPeerId).connectedness == Connectedness.CanConnect
+          ## Now let's do the same but with backoff period
+          await client.disconnectNode(serverRemotePeerInfo)
 
-          # When triggering a reconnection with a backoff period
-          let
-            backoffPeriod = 10.seconds
-            halfBackoffPeriod = 5.seconds
+          waitActive:
+            clientPeerStore.get(serverPeerId).connectedness == Connectedness.CanConnect and
+              serverPeerStore.get(clientPeerId).connectedness == Connectedness.CanConnect
 
+          # When triggering a reconnection with a backoff period
+          let backoffPeriod = chronos.seconds(1)
+          beforeReconnect = getTime().toUnixFloat()
           await client.peerManager.reconnectPeers(WakuRelayCodec, backoffPeriod)
-          await sleepAsync(halfBackoffPeriod)
-
-          # If the backoff period is not over, then the peers should still be marked as CanConnect
-          check:
-            clientPeerStore.get(serverPeerId).connectedness == Connectedness.CanConnect
-            serverPeerStore.get(clientPeerId).connectedness == Connectedness.CanConnect
-
-          # When waiting for the backoff period to be over
-          await sleepAsync(halfBackoffPeriod)
-
-          # Then both peers should be marked as Connected
-          check:
-            clientPeerStore.get(serverPeerId).connectedness == Connectedness.Connected
-            serverPeerStore.get(clientPeerId).connectedness == Connectedness.Connected
-
-        xasyncTest "Automatic Reconnection Implementation (After client restart)":
-          # Given two correctly initialised nodes, that are available for reconnection
-          await server.mountRelay()
-          await client.mountRelay()
-          await client.connectToNodes(@[serverRemotePeerInfo])
-          await server.switch.stop()
-          await client.switch.stop()
-          check:
-            clientPeerStore.get(serverPeerId).connectedness == Connectedness.CanConnect
-            serverPeerStore.get(clientPeerId).connectedness == Connectedness.CanConnect
-
-          # When triggering the reconnection, and some time for the reconnection to happen
-          waitFor allFutures(client.stop(), server.stop())
-          await allFutures(server.start(), client.start())
-          await sleepAsync(FUTURE_TIMEOUT_LONG)
+          let reconnectDurationWithBackoffPeriod =
+            getTime().toUnixFloat() - beforeReconnect
 
           # Then both peers should be marked as Connected
           check:
             clientPeerStore.get(serverPeerId).connectedness == Connectedness.Connected
             serverPeerStore.get(clientPeerId).connectedness == Connectedness.Connected
+            reconnectDurationWithBackoffPeriod >
+              (reconnectDuration + backoffPeriod.seconds.float)
```

Comment:
Not sure I understand why this is true. Shouldn't they be theoretically equal, but in practice any of them be longer than the other one?

Or in other words, why `reconnectDuration + backoffPeriod.seconds.float` can't be greater than `reconnectDurationWithBackoffPeriod`?

Reply:
> Not sure I understand why this is true. Shouldn't they be theoretically equal, but in practice any of them be longer than the other one?
> 
> Or in other words, why `reconnectDuration + backoffPeriod.seconds.float` can't be greater than `reconnectDurationWithBackoffPeriod`?

Actually, in both cases we are doing the same operation but, in when the `backoff` time assigned is != 0, then we add this extra block:
```
    if backoffTime > ZeroDuration:
      debug "Backing off before reconnect",
        peerId = peerInfo.peerId, backoffTime = backoffTime
      # We disconnected recently and still need to wait for a backoff period before connecting
      await sleepAsync(backoffTime)
```
... where we are adding an extra `backoffTime` plus the time spent in the `debug` call plus the if statement. Hence, the `>` condition should happen.

Reply:
Mmm I think that this can be problematic, the only difference is the `if` statement, the log and the time it takes to return from the sleep after the time has passed.

But for example, if the call to `await pm.connectToNodes(@[peerInfo])` in the case without the backoff for some reason takes longer than in the run with the backoff, it can happen that the statement won't be true.

So given that it can be false in certain cases and that I don't think that the check adds much, I would personally remove it. But if not we can keep it and remove it if we see that it gives problems :)

Reply:
Morning! ah yes , makes sense. For now I just simplified the test in https://github.com/waku-org/nwaku/pull/3049/commits/772131ebafe2be1a7465f90087a960c13bf3f0f9
Thanks!